common/src/autogluon/common/model_filter/__init__.py

```diff
@@ -0,0 +1 @@
+from .model_filter import ModelFilter
```

Comment:
rename `model_filter.py` to `_model_filter.py` so that it cleans up import IDE hints if we only care about `ModelFilter`. and can get it here.